snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/tracker/events/ScreenView.java

```diff
@@ -159,4 +242,28 @@ public TrackerPayload getData() {
     public SelfDescribingJson getPayload() {
         return new SelfDescribingJson(TrackerConstants.SCHEMA_SCREEN_VIEW, getData());
     }
+
+    private static String getSnowplowScreenId(Activity activity) {
+        try {
+            Field field = activity.getClass().getField("snowplowScreenId");
+            Object reflectedValue = field.get(activity);
+            if (reflectedValue instanceof String) {
+                return (String)reflectedValue;
+            }
+        } catch (Exception e) {
+            // log here
```

Comment:
Worth using the old logging system here to output to logcat? 
Seems likely that getField might throw a NoSuchFieldException and would be good for developers to know.

Reply:
Right, it's just a silent fault now.